integrations/sentry/main.py

```diff
@@ -16,60 +23,90 @@ class ObjectKind(StrEnum):
     ISSUE_TAG = "issue-tag"
 
 
-def init_client() -> SentryClient:
+def flatten_list(lst: list[Any]) -> list[Any]:
+    return list(chain.from_iterable(lst))
+
+
+def init_client(
+    default_sem: asyncio.Semaphore, issues_sem: asyncio.Semaphore | None = None
+) -> SentryClient:
     sentry_client = SentryClient(
         ocean.integration_config["sentry_host"],
         ocean.integration_config["sentry_token"],
         ocean.integration_config["sentry_organization"],
+        default_sem,
+        issues_sem,
     )
     return sentry_client
 
 
-async def add_tags_to_issue(
-    issue: dict[str, Any], sentry_client: SentryClient
-) -> dict[str, Any]:
-    tags = await sentry_client.get_issue_tags(issue["id"])
-    return {**issue, "__tags": tags}
-
-
 @ocean.on_resync(ObjectKind.PROJECT)
 async def on_resync_project(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    sentry_client = init_client()
-
+    sem = asyncio.Semaphore(MAXIMUM_CONCURRENT_REQUESTS)
+    sentry_client = init_client(sem)
     async for projects in sentry_client.get_paginated_projects():
-        yield projects
+        if projects:
+            logger.info(f"Received {len(projects)} projects")
+            yield projects
 
 
 @ocean.on_resync(ObjectKind.PROJECT_TAG)
 async def on_resync_project_tag(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    sentry_client = init_client()
-
+    sem = asyncio.Semaphore(MAXIMUM_CONCURRENT_REQUESTS)
+    sentry_client = init_client(sem)
     async for projects in sentry_client.get_paginated_projects():
-        for project in projects:
-            tags = await sentry_client.get_project_tags(project["slug"])
-            project_tags = [{**project, "__tags": tag} for tag in tags]
-            yield project_tags
+        logger.info(f"Collecting tags from {len(projects)} projects")
+        project_tags_batch = []
+        tasks = [
+            sentry_client.get_project_tags_iterator(project) for project in projects
+        ]
+        async for project_tags in stream_async_iterators_tasks(*tasks):
+            if project_tags:
+                project_tags_batch.append(project_tags)
+        logger.info(
+            f"Collected {len(project_tags_batch)} project tags from {len(projects)} projects"
+        )
+        yield flatten_list(project_tags_batch)
 
 
 @ocean.on_resync(ObjectKind.ISSUE)
 async def on_resync_issue(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    sentry_client = init_client()
-
-    async for projects in sentry_client.get_paginated_projects():
-        for project in projects:
-            async for issues in sentry_client.get_paginated_issues(project["slug"]):
-                yield issues
+    sem = asyncio.Semaphore(MAXIMUM_CONCURRENT_REQUESTS)
+    issues_sem = asyncio.Semaphore(MAXIMUM_CONCURRENT_REQUESTS_ISSUES)
+    sentry_client = init_client(sem, issues_sem)
```

Comment:
seems like repeated code, can't this be initialized as part of the client, or create an issue client which already holds and initializes those for you, I feel like the  "on_resync" shouldn't even know that this implementation happens ( its an implementation detail of the client )

Reply:
also lets thrive to use full parameters names